src/h5web/visualizations/index.tsx

```diff
@@ -1,101 +1,152 @@
+  Component: ElementType<VisContainerProps>;
+  supportsEntity(entity: HDF5Entity, metadata: HDF5Metadata): boolean;
 }
 
+// @ts-ignore
 export const VIS_DEFS: Record<Vis, VisDef> = {
   [Vis.Raw]: {
     Icon: FiCpu,
-    Component: RawVis,
-    supportsDataset: () => true,
-    defaultMappingState: () => undefined,
+    Component: RawViscontainer,
+    supportsEntity: isDataset,
   },
 
   [Vis.Scalar]: {
     Icon: FiCode,
-    Component: ScalarVis,
-    supportsDataset: (dataset) => {
-      const { type, shape } = dataset;
-      return isBaseType(type) && isScalarShape(shape);
+    Component: ScalarVisContainer,
+    supportsEntity: (entity) => {
+      return (
+        isDataset(entity) &&
+        isBaseType(entity.type) &&
+        isScalarShape(entity.shape)
+      );
     },
-    defaultMappingState: () => undefined,
   },
 
   [Vis.Matrix]: {
     Icon: FiGrid,
-    Component: MappedMatrixVis,
-    supportsDataset: (dataset) => {
-      const { type, shape } = dataset;
-      return isBaseType(type) && isSimpleShape(shape) && shape.dims.length >= 1;
-    },
-    defaultMappingState: (dataset) => {
-      const { dims } = dataset.shape as HDF5SimpleShape;
-      return dims.length === 1
-        ? ['x']
-        : [...range(dims.length - 2).fill(0), 'y', 'x'];
+    Component: MatrixVisContainer,
+    supportsEntity: (entity) => {
+      return (
+        isDataset(entity) &&
+        isBaseType(entity.type) &&
+        isSimpleShape(entity.shape) &&
+        entity.shape.dims.length >= 1
+      );
     },
   },
 
   [Vis.Line]: {
     Icon: FiActivity,
     Toolbar: LineToolbar,
-    Component: MappedLineVis,
-    supportsDataset: (dataset) => {
-      const { type, shape } = dataset;
+    Component: LineVisContainer,
+    supportsEntity: (entity) => {
       return (
-        isNumericType(type) && isSimpleShape(shape) && shape.dims.length >= 1
+        isDataset(entity) &&
+        isNumericType(entity.type) &&
+        isSimpleShape(entity.shape) &&
+        entity.shape.dims.length >= 1
       );
     },
-    defaultMappingState: (dataset) => {
-      const { dims } = dataset.shape as HDF5SimpleShape;
-      return [...range(dims.length - 1).fill(0), 'x'];
-    },
   },
 
   [Vis.Heatmap]: {
     Icon: FiMap,
     Toolbar: HeatmapToolbar,
-    Component: MappedHeatmapVis,
-    supportsDataset: (dataset) => {
-      const { type, shape } = dataset;
+    Component: HeatmapVisContainer,
+    supportsEntity: (entity) => {
+      return (
+        isDataset(entity) &&
+        isNumericType(entity.type) &&
+        isSimpleShape(entity.shape) &&
+        entity.shape.dims.length >= 2
+      );
+    },
+  },
+
+  [Vis.NxSpectrum]: {
+    Icon: FiActivity,
+    Toolbar: LineToolbar,
+    Component: NxSpectrumContainer,
+    supportsEntity: (entity, metadata) => {
+      if (!isGroup(entity) || !isNxDataGroup(entity)) {
+        return false;
+      }
+
+      const dataset = getSignalDataset(entity, metadata);
```

Comment:
I need the `metadata` object to find the signal dataset, which is why the signature of `supportsEntity` now has a `metadata` argument.